File: services/exercise_compiler/src/tmp/tmp.cpp

// Type your code here
#include <stdio.h>

 void quicksort(int array[], int start, int end) {
    if (start < end) {
        int pivot = array[start];
        int i = start;
        int j = end;
        int tmp;

        while (i < j) {
            while (array[i] <= pivot) {
                i++;
            }
            while (array[j] > pivot) {
                j--;
            }
            if (i < j) {
                tmp = array[i];
                array[i] = array[j];
                array[j] = tmp;
            }
        }
        tmp = array[j];
        array[j] = array[start];
        array[start] = tmp;
        quicksort(array, start, j - 1);
        quicksort(array, j + 1, end);
    }
}

int main() {
    int array[] = {10, 5, 2, 3};
    int size = sizeof(array) / sizeof(array[0]);
    quicksort(array, 0, size - 1);
    for (int i = 0; i < size; i++) {
        printf("%d ", array[i]);
    }
    printf("\n");
    return 0;
}

File: services/exercise_compiler/src/tmp/tmp.go

// Type your code here
package main

import "fmt"

func quicksort(array []int) []int {
    if len(array) < 2 {
        return array
    } else {
        pivot := array[0]
        var less []int
        var greater []int
        for _, i := range array[1:] {
            if i <= pivot {
                less = append(less, i)
            } else {
                greater = append(greater, i)
            }
        }
        less = quicksort(less)
        greater = quicksort(greater)
        return append(append(less, pivot), greater...)
    }
}

func main() {
    array := []int{10, 5, 2, 3}
    fmt.Println(quicksort(array))
}